### DISCUSSION TITLE
Migrate from weppy to emmett [uploads]

### DISCUSSION THREAD

**Discussion by josejachuf** [emmett-framework/emmett#488] [Q&A]
I already solved this. The solution is strange, at least I don't understand.

I will try to explain the steps I made:

1) I moved the uploads directory to the new server with Emmett
2) Assign **www** as owner the entire directory: `chown -R www:www /www`
3) I try to download any of the files and get a 404. I look and the file exists
4) I try an upload and I get a 500 error (internal error)

```
[2024-02-13 21:28:04,393] ERROR in handlers: Application exception:
Traceback (most recent call last):

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/www/venv/lib/python3.11/site-packages/emmett/forms.py", line 416, in _process
    newfilename = field.store(
                  ^^^^^^^^^^^^
  File "/www/venv/lib/python3.11/site-packages/pydal/objects.py", line 1706, in store
    os.makedirs(path)
  File "<frozen os>", line 225, in makedirs
FileExistsError: [Errno 17] File exists: '/../uploads'
```

5) I change the owner again `chown -R www:www /www`
6) I try upload again and now it works fine.
7) I try download again and now it works fine.

Why did not recognize the "Upload" directory if it already existed and had the corresponding permits and tried to create it in the first upload?

I don't understand what happened

**Comment by josejachuf** [comment]
This could be because at the time of starting the app, the 'uploads' folder didn't exist yet, and then I copied it while the app was running? And that's why it didn't recognize the existence of 'uploads'.

**Reply by gi0baro** [reply]
@josejachuf I'm not sure on what the exact issue was. Unfortunately this still rely on pyDAL code and Emmett is still dependant on a very old version of pyDAL due to incompatible changes.

I might investigate this a bit, but since you managed to get this working I consider it as a low priority thing.
